algorithms/bug2.py

```diff
@@ -0,0 +1,291 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+import os
+import sys
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+
+from tools.map import Map
+
+
+max_index = -1
+
+
+def reached_goal(grid, x, y):
```

Comment:
If you are using these functions in each file, then this should be extracted to a separate file and imported. Same with other functions in this file (if_boundary, etc.).

Reply:
I now put all the common functions in a file called helper.py